DOI.js

```diff
@@ -93,52 +94,83 @@ function completeDOIs(doc) {
 	}
 }
 
-function retrieveDOIs(DOIs, doc) {
-	__num_DOIs = DOIs.length;
 
+var providers = [
+	"11645bd1-0420-45c1-badb-53fb41eeb753", //CrossRef
+	"9f1fb86b-92c8-4db7-b8ee-0b481d456428"  //DataCite
+];
+
+
+/*
+var providers = [
+	"9f1fb86b-92c8-4db7-b8ee-0b481d456428", //DataCite
+	"11645bd1-0420-45c1-badb-53fb41eeb753"  //CrossRef
+];
+*/
+
+function retrieveDOIs(DOIs, doc, provider) {
+	__num_DOIs = DOIs.length;
+	
 	for(var i=0, n=DOIs.length; i<n; i++) {
 		(function(doc, DOI) {
 			var translate = Zotero.loadTranslator("search");
-			translate.setTranslator("11645bd1-0420-45c1-badb-53fb41eeb753");
-	
+			translate.setTranslator(provider);
+			
 			var item = {"itemType":"journalArticle", "DOI":DOI};
 			translate.setSearch(item);
-	
+
 			// don't save when item is done
 			translate.setHandler("itemDone", function(translate, item) {
 				if (!item.title) {
-					Zotero.debug("No title available for " + DOI);
-					return;
+					Zotero.debug("No title available for " + item.DOI);
+					item.title = "[No Title]";
+					//return;
 				}
-				
-				item.repository = "CrossRef";
-				items[DOI] = item;
-				selectArray[DOI] = item.title;
+
+				items[item.DOI] = item;
+				selectArray[item.DOI] = item.title;
+				// done means not remaining anymore
+				Z.debug(item.DOI + "   " + remainingDOIs + "   " + provider + "    " + remainingDOIs.indexOf(item.DOI) );
+				remainingDOIs.splice(remainingDOIs.indexOf(item.DOI), 1);
```

Comment:
Keep in mind that `item.DOI` may not be the same as the DOI that was passed into the translator. It's an ok hack for now, but I wouldn't rely on this 100%. Make sure this doesn't fail with an error. Also, this could mean that we retrieve an item twice. If there's a mismatch during CrossRef lookup, we would look it up again on DataCite. I think the same DOI should not exist in both DataCite and CrossRef repos, but I'm not sure.